### DISCUSSION TITLE
How to better review the upgrades done by pnpm install

### DISCUSSION THREAD

**Discussion by jeremymeng** [pnpm/pnpm#4748] [Q&A]
We use rush and pnpm to manage dependencies in our monorepo.  Currently we review the package upgrades by looking at the diffs of the pnpm-lock.yaml file for example in this PR. https://github.com/Azure/azure-sdk-for-js/pull/21886/files.  However the file is large and there are many upgrades of the same.

Are there better way to review the package upgrades/removal/etc.?

**Comment by jeremymeng** [comment]
any ideas?

**Reply by orgrimarr** [reply]
Clearly not the best solution, but works for me.

I wrote a script that :

On both commits to diff

- git checkout commitHash **/package.json pnpm-lock.yaml
- pnpm i
- pnpm ls --depth 0 -r --json

Then diff the result of both pnpm list result, and pretty print it
